src/middlewares/preInvokeMiddleware.tsx

```diff
@@ -0,0 +1,34 @@
+/**
+ * Process preInvoke operation before action sendOperation
+ */
+
+import {AnyAction, Dispatch, Middleware, MiddlewareAPI} from 'redux'
+import {$do, types} from '../actions/actions'
+import {Store as CoreStore} from '../interfaces/store'
+
+const preInvokeAction = ({getState, dispatch}: MiddlewareAPI<Dispatch<AnyAction>, CoreStore>) => (next: Dispatch) =>
+    (action: AnyAction) => {
+        if (action.type === types.sendOperation) {
+            const bcName = action.payload.bcName
+            const operationType = action.payload.operationType
+            const widgetName = action.payload.widgetName
+            const preInvoke = action.payload?.confirmOperation
+
+            return preInvoke && !action.payload.confirm
```

Comment:
You probably do not need `action.payload.confirm` here

Reply:
It necessary that `preInvokeMiddleware`  not execute when confirm response is received